### ISSUE TITLE
std::result_of is removed in C++20

### ISSUE THREAD

**Issue by robindegen** [google/googletest#2914]
std::result_of is removed in C++20, causing gmock-actions.h:819 to fail in Visual Studio 2019 16.6.2 with C++20 enabled. However std::invoke_result is available since C++17.

For C++20:
```
  using ReturnType = typename std::result_of<MethodPtr(Class*)>::type;
```
should be replaced with:
```
  using ReturnType = typename std::invoke_result_t<MethodPtr, Class*>;
```

**Comment by sbenzaquen** [comment]
That code has already been replaced in https://github.com/google/googletest/commit/61f010d703b32de9bfb20ab90ece38ab2f25977f

**Comment by robindegen** [comment]
@sbenzaquen Is there any timeline for a release that contains this fix?

**Comment by Andreas-Schniertshauer** [comment]
Is there any plan to release a new version of googletest in the near future where this problem is solved?

**Comment by milbrandt** [comment]
As C++20 is now officially approved it would be time to have that released.

**Comment by twhiting** [comment]
Hi, also adding to this. Waiting on this release w/ the fix.

**Comment by jwmcglynn** [comment]
I had this issue and resolved it after upgrading to v1.11.0, so I think the fix landed in that release.
